### ISSUE TITLE
build breaks with SIMD in mask4

### ISSUE THREAD

**Issue by cdc100** [OpenImageIO/oiio#1340]
It seems commit 861af905239f4b114e7b6519f97cdcbf6102febf breaks the builds when using SIMD.
```
In file included from /home/cdc100/deploy/oiio/src/include/OpenImageIO/fmath.h:64:0,
                 from /home/cdc100/deploy/oiio/src/include/OpenImageIO/hash.h:50,
                 from /home/cdc100/deploy/oiio/src/include/OpenImageIO/strutil.h:54,
                 from /home/cdc100/deploy/oiio/src/libutil/argparse.cpp:43:
/home/cdc100/deploy/oiio/src/include/OpenImageIO/simd.h: In static member function ‘static const OpenImageIO::v1_7::simd::mask4 OpenImageIO::v1_7::simd::mask4::True()’:
/home/cdc100/deploy/oiio/src/include/OpenImageIO/simd.h:284:65: error: cannot convert ‘__m128i {aka __vector(2) long long int}’ to ‘__m128 {aka __vector(4) float}’ for argument ‘1’ to ‘__m128i _mm_castps_si128(__m128)’
         __m128i anyval = _mm_castps_si128 (_mm_undefined_si128());
                                                                 ^
/home/cdc100/deploy/oiio/src/include/OpenImageIO/simd.h: In static member function ‘static const OpenImageIO::v1_7::simd::int4 OpenImageIO::v1_7::simd::int4::NegOne()’:
/home/cdc100/deploy/oiio/src/include/OpenImageIO/simd.h:657:65: error: cannot convert ‘__m128i {aka __vector(2) long long int}’ to ‘__m128 {aka __vector(4) float}’ for argument ‘1’ to ‘__m128i _mm_castps_si128(__m128)’
         __m128i anyval = _mm_castps_si128 (_mm_undefined_si128());  
```
Disabling SIMD via **USE_SIMD=0** results in another error
```
/home/cdc100/deploy/oiio/src/include/OpenImageIO/simd.h: In member function ‘void OpenImageIO::v1_7::simd::float4::load(const half*)’:
/home/cdc100/deploy/oiio/src/include/OpenImageIO/simd.h:1681:9: error: ‘m_vec’ was not declared in this scope
         m_vec = _mm_cvtph_ps (a);
         ^
In file included from /home/cdc100/deploy/oiio/src/include/OpenImageIO/fmath.h:64:0,
                 from /home/cdc100/deploy/oiio/src/include/OpenImageIO/hash.h:50,
                 from /home/cdc100/deploy/oiio/src/include/OpenImageIO/strutil.h:54,
                 from /home/cdc100/deploy/oiio/src/include/OpenImageIO/ustring.h:137,
                 from /home/cdc100/deploy/oiio/src/libutil/typedesc.cpp:39:
/home/cdc100/deploy/oiio/src/include/OpenImageIO/simd.h: In member function ‘void OpenImageIO::v1_7::simd::float4::store(half*) const’:
/home/cdc100/deploy/oiio/src/include/OpenImageIO/simd.h:1784:35: error: ‘m_vec’ was not declared in this scope
         __m128i h = _mm_cvtps_ph (m_vec, (_MM_FROUND_TO_NEAREST_INT |_MM_FROUND_NO_EXC));   
```
The references i can find point to a similar issue with ticket #952

**Comment by lgritz** [comment]
Proposed fix in #1342. Can you give that a try and see if it builds for you?

**Comment by cdc100** [comment]
@lgritz pull request #1342 fixed it, it builds fine now.
